packages/fxa-auth-server/lib/routes/validators.js

```diff
@@ -384,16 +384,48 @@ module.exports.subscriptionsSubscriptionListValidator = isA.object({
 // - metadata can contain arbitrary keys that we don't expect (e.g. used by other systems)
 // - but we can make a good effort at validating what we expect to see when we see it
 module.exports.subscriptionPlanMetadataValidator = isA.object().unknown(true);
+
+const privacyPattern = /^product:privacyNoticeURL/;
+const privacyDownloadPattern = /^product:privacyNoticeDownloadURL/;
+const tosPattern = /^product:termsOfServiceURL/;
+const tosDownloadPattern = /^product:termsOfServiceDownloadURL/;
+const capabilitiesClientIdPattern = /^capabilities/;
+const legalResourceDomainPattern = /^https:\/\/accounts-static\.cdn\.mozilla\.net\/legal\/(.*)/;
+
 module.exports.subscriptionProductMetadataValidator = isA
   .object({
+    webIconURL: isA.string().uri().required(),
+    upgradeCTA: isA.string().optional(),
+    downloadURL: isA.string().uri().required(),
+    appStoreLink: isA.string().uri().optional(),
+    playStoreLink: isA.string().uri().optional(),
     productSet: isA.string().optional(),
     productOrder: isA.number().optional(),
-    iconURL: isA.string().optional(),
-    upgradeCTA: isA.string().optional(),
-    downloadURL: isA.string().optional(),
-    appStoreLink: isA.string().optional(),
-    playStoreLink: isA.string().optional(),
   })
+  .with('productOrder', 'productSet')
```

Comment:
@meandavejustice Hey sorry but we were wrong about this.  We want allow a plan to inherit a productSet from the product but not have a productOrder set on it.

Reply:
updated